### ISSUE TITLE
Ability to document class instances better

### ISSUE THREAD

**Issue by michaelBenin** [jsdoc3/jsdoc#949]
Example:

```javascript
/**
 * @classinstance
 * @classdesc validates an object
 * @property {Function} run - executes the validation, returns a promise
 * @throws Checkit.Error
 * @returns {Promise}
 * @see {@link https://github.com/tgriesser/checkit#checkitrun}
 */
var validate = new Checkit({})
```

**Comment by hegemonic** [comment]
Why not just use `@type {Checkit}`, which indicates that `validate` is a `Checkit` instance?

**Comment by michaelBenin** [comment]
<inner> validate :Checkit
Output:

validates an object

Type:

Checkit
Properties:

Name	Type	Description
run	function	
executes the validation, returns a promise

Ignores the returns, throws, see. 

Is there anyway to add those?

**Comment by hegemonic** [comment]
I took a closer look at your example. I'm not sure the docs you have right now are correct--presumably it's the `Checkit` instance's `run` method, and not the instance itself, that returns a promise.

The normal way to document this in JSDoc-land would be to just document `Checkit` (including its `run` method) and indicate that `validate`'s type is `Checkit`, as I suggested. Something like this:

```js
/**
 * Checkit is used for validation.
 * @class
 * @see {@link https://github.com/tgriesser/checkit}
 */
var Checkit = require('checkit');

/**
 * Run the validator.
 * @function Checkit#run
 * @param {Object} obj - The object to validate.
 * @returns {Promise}
 * @see {@link https://github.com/tgriesser/checkit#checkitrun}
 */

/**
 * Validator.
 * @type {Checkit}
 */
var validate = new Checkit({});
```

If I understand correctly, you want JSDoc to automatically show that `validate` has a method called `run`, and to display all the docs for that method in the same place as `validate`. It's true that, at the moment, there is no convenient way to do that for class instances. I'm not sure a `@classinstance` tag is the best way to solve the problem, but I think I get what you're asking for now. (If I'm incorrect, please let me know.)

**Comment by michaelBenin** [comment]
This is exactly what I'm looking for. 

Also, after running the example above, unfortunately I'm still not seeing the param, returns, and see.

**Comment by hegemonic** [comment]
Yes, that's because nobody has implemented this feature request yet. :)
